### PR TITLE
EN-10778 - small refactor on configs

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by raduchis — APPROVED** [review]

File: dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go [hunk 1]
```diff
@@ -278,7 +284,8 @@ func (brcf *baseResolversContainerFactory) createMiniBlocksResolver(
 func (brcf *baseResolversContainerFactory) generatePeerAuthenticationResolver() error {
 	identifierPeerAuth := common.PeerAuthenticationTopic
 	shardC := brcf.shardCoordinator
-	resolverSender, err := brcf.createOneResolverSender(identifierPeerAuth, EmptyExcludePeersOnTopic, shardC.SelfId())
+	numIntraShardPeers := brcf.numTotalPeers - brcf.numCrossShardPeers
```
Comment (bogdan-rosianu) [inline dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go:287]: maybe make `numIntraShardPeers` part of the `baseResolversContainerFactory`? you do the subtraction a lot of times, the same way

File: dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go [hunk 2]
```diff
@@ -309,34 +316,17 @@ func (brcf *baseResolversContainerFactory) generatePeerAuthenticationResolver()
...
-
 func (brcf *baseResolversContainerFactory) createOneResolverSenderWithSpecifiedNumRequests(
 	topic string,
 	excludedTopic string,
 	targetShardId uint32,
-	numCrossShard int,
-	numIntraShard int,
-	numFullHistory int,
-	currentNetworkEpochProvider dataRetriever.CurrentNetworkEpochProviderHandler,
+	numCrossShardPeers int,
```
Comment (iulianpascalau) [inline dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go:323]: why do we have to provide the numCrossShardPeers value?
Please check if it is always passed as brcf.numCrossShardPeers and if yes, might remove the parameter

File: dataRetriever/factory/resolverscontainer/metaResolversContainerFactory_test.go [hunk 3]
```diff
@@ -213,6 +214,83 @@ func TestNewMetaResolversContainerFactory_NilTrieDataGetterShouldErr(t *testing.
...
+
+	args := getArgumentsMeta()
+	args.ResolverConfig.NumCrossShardPeers = 0
+	rcf, err := resolverscontainer.NewMetaResolversContainerFactory(args)
+
+	assert.Nil(t, rcf)
+	assert.True(t, errors.Is(err, dataRetriever.ErrInvalidValue))
+}
+
+func TestNewMetaResolversContainerFactory_InvalidNumTotalPeersShouldErr(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/factory/resolverscontainer/metaResolversContainerFactory_test.go:261]: added

File: dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go (hunk 2 above)
Comment (sstanculeanu) [reply dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go:321]: `generateTrieNodesResolvers` methods call it with 0 `numCrossShardPeers` and all `numTotalPeers` intra

**Review by bogdan-rosianu — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
New errors and warnings are safe. 

@@                            Log scanner                                  @@
##                            EN-10778-config-refactor                     ##
================================================================================
- Known Warnings              14
- New Warnings                5
- Known Errors                0
- New Errors                  1
- Panics                      0
================================================================================

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**